### ISSUE TITLE
Ethernet does not work (TEP0700IMX7)

### ISSUE THREAD

**Issue by dbeulen** [TechNexion/meta-tn-imx-bsp#11]
We have 6 displays where ethernet does not work using the sample builds. We tried building using yocto (https://developer.technexion.com/docs/building-an-image-with-yocto)  as well as the prebuild images (https://download.technexion.com/demo_software/TEP/IMX7/tep0700-imx7d/DiskImage/) yocto and ubuntu. All these images have the same problem. the lights at the ethernet port do not turn on. Both the POE port as the normal port do not work.

On the preinstalled software the ethernet does work, so it is not a hardware issue.

After a long research we found out our displays might be a different hardware version (A1). This might be the reason it does not work, but we cannot say for sure we have the A1 version.

Details of one of the displays:
Part No: S600406010040480000511
SN number: 17202213400018240005
Model name: TEP0700IMX7DR10E04LPOE

TLDR 2 questions:

1. How can we tell if we have A1 version?
2. What can we try to fix this issue?

**Comment by richard-hu** [comment]
@dbeulen :

I apologize for the trouble caused.
We are trying to replicate the issue on our side to figure out what seems to be the problem.

I will keep updating the status.

BR,

Richard

**Comment by richard-hu** [comment]
@dbeulen :

Our test team confirms that we can replicate this issue.
Our software RD is delving into it.

BR,

Richard

**Comment by richard-hu** [comment]
@dbeulen:

I create an internal ticket for this issue already.
Could you offer your company and order information to fae@technexion.com ?
We will contact you via fae@technexion.com .
Github seems not a good place to reveal company or personal information.

Thanks~

BR,

Richard

**Comment by dbeulen** [comment]
Hello Richard,

Thank you for your reply, I have sent an email to the provided email address.

**Comment by richard-hu** [comment]
@dbeulen 

I tweak u-boot a little bit to add support TEP1-IMX7D rev.A1.
https://github.com/TechNexion/u-boot-tn-imx/commit/236a9e04ef3656fc1df94bb2e916dba497414e46
https://github.com/TechNexion/u-boot-tn-imx/commit/3defa8fe4bcc524d9b7bd47df08ec3369b189535
https://github.com/TechNexion/u-boot-tn-imx/commit/dd83055c2348915626ed6bdbb9c8d372550c6c31

Ethernet works fine now.

I also update the u-boot version to latest one in TechNexion BSP meta-layer:
https://github.com/TechNexion/meta-tn-imx-bsp/commit/de9e111145d8fc691db1d82417c3e1f3eb615f3d

Could you fetch Yocto 3.3 `-next` branch and create the build again?

Please refer to https://github.com/TechNexion/tn-imx-yocto-manifest/tree/hardknott_5.10.y-next
```
repo init -u https://github.com/TechNexion/tn-imx-yocto-manifest.git -b hardknott_5.10.y-next -m imx-5.10.72-2.2.0.xml
repo sync -j8
```

BR,

Richard

**Comment by dbeulen** [comment]
Hy Richard,

Thank you for your support. I can confirm that ethernet is working now on the hardknott_5.19.y-next branch!

Greetings,
Denny
